### PR TITLE
KOGITO-3156 fix integration tests broken for Quarkus

### PR DESCRIPTION
**Pull request by tiagodolphine**
https://issues.redhat.com/browse/KOGITO-3156

The Quarkus integration test was broken because of the lifecycle order the test resources are stated, the `KogitoServiceRandomPortQuarkusTestResource` should be started before the `JobServiceQuarkusTestResource` because the         `Testcontainers.exposeHostPorts(httpPort);` is called there and it is necessary to be called before starting the jobs-service container.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Review by danielezonca — APPROVED** [review]
LGTM :+1:

**Comment by tiagodolphine** [comment]
jenkins retest this

File: integration-tests/integration-tests-common/src/main/java/org/kie/kogito/testcontainers/JobServiceContainer.java [hunk 1]
```diff
@@ -36,11 +36,13 @@
...
     private static final Logger LOGGER = LoggerFactory.getLogger(JobServiceContainer.class);
 
-    public JobServiceContainer() {
+    @Override
+    public void start() {
         addExposedPort(PORT);
         withLogConsumer(new Slf4jLogConsumer(LOGGER));
         waitingFor(Wait.forLogMessage(".*Listening on:.*", 1));
         setDockerImageName(getImageName());
+        super.start();
```
Comment (Sgitario) [inline integration-tests/integration-tests-common/src/main/java/org/kie/kogito/testcontainers/JobServiceContainer.java]: Is this part of the fix? It seems unrelated.

File: integration-tests/integration-tests-quarkus/src/test/java/org/kie/kogito/resources/KogitoServiceRandomPortQuarkusTestResource.java [hunk 2]
```diff
@@ -25,6 +25,13 @@ public KogitoServiceRandomPortQuarkusTestResource() {
         super(new KogitoServiceRandomPortTestResource());
     }
 
+    @Override
+    public int order() {
+        //Guarantee it will be executed before other test resources that depends on this to set the
+        //Testcontainers.exposeHostPorts(httpPort), see KogitoServiceRandomPortTestResource.start()
+        return Integer.MIN_VALUE;
```
Comment (Sgitario) [inline integration-tests/integration-tests-quarkus/src/test/java/org/kie/kogito/resources/KogitoServiceRandomPortQuarkusTestResource.java]: What about we want to integrate jobs service with data index or infinispan? I think we should think a value here where we can still configure all the containers accordingly.
Also, we can do this once we need it.

Comment (tiagodolphine) [reply integration-tests/integration-tests-quarkus/src/test/java/org/kie/kogito/resources/KogitoServiceRandomPortQuarkusTestResource.java]: yeah, we could have some way to control the order the test resources are initialized, for now, the only restriction we have when setting the Testcontainers.exposeHostPorts it should be done before the container starts.

**Comment by tiagodolphine** [comment]
It seems the changes didn't fix the issue on Jenkins, but locally it is working fine, I need to check it again.

File: integration-tests/integration-tests-common/src/main/java/org/kie/kogito/it/KogitoServiceRandomPortTestResource.java [hunk 3]
```diff
@@ -32,17 +32,25 @@ public String getResourceName() {
...
-        //It should be called be fore the container is instantiated
         Testcontainers.exposeHostPorts(httpPort);
         //the hostname for the container to access the host is "host.testcontainers.internal"
         //https://www.testcontainers.org/features/networking/#exposing-host-ports-to-the-container
         System.setProperty(KOGITO_SERVICE_URL, "http://host.testcontainers.internal:" + httpPort);
     }
 
+    @Override
+    public void start() {
+
```
Comment (danielezonca) [inline integration-tests/integration-tests-common/src/main/java/org/kie/kogito/it/KogitoServiceRandomPortTestResource.java:51]: If we want to prevent `super.start()` execution, what about a comment here to mention the reason?

Comment (tiagodolphine) [reply integration-tests/integration-tests-common/src/main/java/org/kie/kogito/it/KogitoServiceRandomPortTestResource.java:51]: the start() needs to be implemented because it is from the interface TestResource, basically, there is no super.start(), I moved the initialization of the port to the constructor inserted of the start() method.

File: integration-tests/integration-tests-common/src/main/java/org/kie/kogito/testcontainers/JobServiceContainer.java (hunk 1 above)
Comment (tiagodolphine) [reply integration-tests/integration-tests-common/src/main/java/org/kie/kogito/testcontainers/JobServiceContainer.java]: rolled back since it didn't work.

File: integration-tests/integration-tests-quarkus/src/test/java/org/kie/kogito/resources/KogitoServiceRandomPortQuarkusTestResource.java (hunk 2 above)
Comment (tiagodolphine) [reply integration-tests/integration-tests-quarkus/src/test/java/org/kie/kogito/resources/KogitoServiceRandomPortQuarkusTestResource.java]: rolled back since it didn't work.

**Comment by tiagodolphine** [comment]
it seems now the build has failed but due to unrelated an issue on kogito-examples

**Comment by tiagodolphine** [comment]
jenkins retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by ruromero** [comment]
jenkins retest this

**Comment by tiagodolphine** [comment]
the build is failing, but due to an unrelated issue `Kogito Example :: Process Infinispan Persistence Quarkus FAILURE`, any clue on that? @danielezonca @cristianonicolai

**Comment by ruromero** [comment]
The build is green now, the fixes were merged in https://github.com/kiegroup/kogito-runtimes/pull/727
I think this one is good to go

**Comment by tiagodolphine** [comment]
great, thanks @ruromero
